Launchpad: Fix domain copy icon spacing

edanzer opened the pull request:
Resolves: https://github.com/Automattic/wp-calypso/issues/75720

### Proposed Changes

Fixes some small spacing issues with the copy icon in the Launchpad > domain name box. The icon is appearing too close to the Customize badge when domain names are long. Note that we only show the customize badge now for newsletters, so you would not see this issue on other flows. 

<img width="806" alt="copy icon" src="https://user-images.githubusercontent.com/21228350/236952267-69112148-6048-48a5-a65e-7edd7d6ccc65.png">

### Testing Instructions

1. Checkout this branch and run yarn and yarn start if needed. 
2. Start a newsletter site at http://calypso.localhost:3000/setup/newsletter/intro and use a very long domain name. You can also use an existing, launchpad-enabled newsletter site, and just update the domain name. 
3. Go to launchpad at http://calypso.localhost:3000/setup/newsletter/launchpad?siteSlug=SITESLUG. If you don't have a long domain name, you can just update the domain name on the screen via the developer tools. 
4. Confirm that the icon always fits well with comfortable spacing.

edanzer committed:
Fix launchpad copy icon spacing

simison commented on client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/style.scss:
```diff
@@ -211,20 +211,23 @@
 	.launchpad__url-box-domain:hover > .launchpad__clipboard-button {
 		opacity: 1;
+		margin-top: 2px;
+		flex-shrink: 0;
```
I would add `:hover` to the rule below ([line 233](https://github.com/Automattic/wp-calypso/pull/76689/files#diff-3f2ed61b57c0fdab5f9b1914740a42b1352e8035d490bc5a318d6950c639197dR233)) and move these there. That way styling applies both on hover and on focus:

<img width="426" alt="Screenshot 2023-05-09 at 10 34 13" src="https://user-images.githubusercontent.com/87168/237027049-1ea77d68-f6fb-4560-bfdf-1fac26e51ec9.png">

edanzer replied:
Address this. See https://github.com/Automattic/wp-calypso/pull/76689#issuecomment-1540686143

simison reviewed (approved):
Thanks for the fix! Tests well on Firefox desktop/mobile screens, small and long URLs.

Left a small note; good to merge after that.

simison commented:
Looking closer, is there a 1px jump down on mobile version?

edanzer committed:
Tweak css rules for copy icon

edanzer commented:
@simison - I addressed comments above together. I moved the new css rules to the generic `.launchpad__clipboard-button selector` rather than applying them only on hover. This fixes the small 2px jump you were seeing hover and ensures consistency for hover and focus. 

I also just consolidated the hover/focus rules for clarity and DRY.
